compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt

```diff
@@ -618,4 +618,226 @@ class InterfaceMergerTest(
+      interface ContributingInterface
+      
+      @ContributesTo(Any::class, replaces = [ContributingInterface::class])
+      interface SecondContributingInterface
+      
+      $annotation(Any::class)
+      interface ComponentInterface
+      
+      $annotation(Unit::class)
+      interface SubcomponentInterface
+      """
+    ) {
+      assertThat(componentInterface extends contributingInterface).isFalse()
+      assertThat(componentInterface extends secondContributingInterface).isTrue()
+      assertThat(subcomponentInterface extends contributingInterface).isTrue()
+      assertThat(subcomponentInterface extends secondContributingInterface).isFalse()
+    }
+  }
+
+  @Test
+  fun `replaced interfaces contributed to multiple scopes must use the same scope`() {
+    assumeKotlin16()
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+      
+      import com.squareup.anvil.annotations.ContributesTo
+      $import
+      
+      @ContributesTo(Any::class)
+      @ContributesTo(Unit::class)
+      interface ContributingInterface
+      
+      @ContributesTo(Int::class, replaces = [ContributingInterface::class])
+      interface SecondContributingInterface
+      
+      $annotation(Any::class)
+      interface ComponentInterface
+      
+      $annotation(Unit::class)
+      interface SubcomponentInterface1
+      
+      $annotation(Int::class)
+      interface SubcomponentInterface2
+      """
+    ) {
+      assertThat(exitCode).isError()
+      assertThat(messages).contains(
+        "com.squareup.test.SecondContributingInterface with scope kotlin.Int wants to " +
+          "replace com.squareup.test.ContributingInterface, but the replaced class isn't " +
+          "contributed to the same scope."
+      )
+    }
+  }
+
+  @Test fun `interfaces contributed to multiple scopes can be excluded in one scope`() {
+    assumeKotlin16()
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+      
+      import com.squareup.anvil.annotations.ContributesTo
+      $import
+      
+      @ContributesTo(Any::class)
+      @ContributesTo(Unit::class)
+      interface ContributingInterface
+      
+      @ContributesTo(Any::class)
+      interface SecondContributingInterface
+      
+      $annotation(Any::class, exclude = [ContributingInterface::class])
+      interface ComponentInterface
+      
+      $annotation(Unit::class)
+      interface SubcomponentInterface
+      """
+    ) {
+      assertThat(componentInterface extends contributingInterface).isFalse()
+      assertThat(componentInterface extends secondContributingInterface).isTrue()
+      assertThat(subcomponentInterface extends contributingInterface).isTrue()
+      assertThat(subcomponentInterface extends secondContributingInterface).isFalse()
+    }
+  }
+
+  @Test fun `contributed interfaces in the old format are picked up`() {
+    val result = AnvilCompilation()
+      .useIR(USE_IR)
+      .enableAnvil(false)
```

Comment:
Also curious if maybe `enableAnvil` should be called something else; at least, when I read this test and see that we are disabling Anvil for AnvilCompilation, but the test is for contributed interfaces being picked up, I would expect nothing to be picked up since Anvil was disabled. So perhaps this is a misunderstanding on my part